checker.go

```diff
@@ -73,6 +73,34 @@ func (c *equalsChecker) Check(got interface{}, args []interface{}, note func(key
 	return nil
 }
 
+// NilError is a Checker that checks whether an error value is nil.
+// For instance:
+//
+//    c.Assert(err, qt.NilError)
+//
+// This should be used in preference to qt.IsNil, because
+// that can succeed inappropriately when an error value
+// contains a concrete type that is itself nil.
+// (See https://golang.org/doc/faq#nil_error)
+var NilError Checker = &nilErrorChecker{
+	argNames: []string{"error value"},
+}
+
+type nilErrorChecker struct {
+	argNames
+}
+
+// Check implements Checker.Check by checking that got is a nil error.
+func (c *nilErrorChecker) Check(got interface{}, args []interface{}, note func(key string, value interface{})) (err error) {
+	if got == nil {
+		return nil
+	}
+	if _, ok := got.(error); !ok {
+		return BadCheckf("value is not an error")
+	}
+	return errors.New("got non-nil error")
+}
+
```

Comment:
What do you think about hinting using this in the Equals checker when got is an error and want is nil?
We could add a hint with a note() for example.